File: 30-Days-Of-Code/Day 08/dictandmap.py
n = int(input().strip())

numbers = {}

# populate
for _ in range(n):
    entry = input().strip().split(" ")
    numbers[entry[0]] = entry[1]

# Until no more queries
while True:
    query = input()
    if query == "":
        break
    elif query in numbers:
        print("{}={}".format(query, numbers[query]))
    else:
        print("Not found")

File: 30-Days-Of-Code/Day 08/dictandmap.cpp
#include <iostream>
#include <map>

using namespace std;

int main() {
    // Number of entries
    int n;
    cin >> n;

    map<string, string> phoneNumbers;

    // Populate
    for (int i = 0; i < n; i++) {
        string name;
        string number;
        cin >> name;
        cin >> number;
        phoneNumbers[name] = number;
    }

    string query;

    // flush cin
    cin.ignore();
    while (getline(cin, query)) {
        // if no entry
        if (query == "") {
            break;
        }
        // if not found
        else if (phoneNumbers.find(query) == phoneNumbers.end()) {
            cout << "Not found" << endl;
        } else {
            cout << query << "=" << phoneNumbers[query] << endl;
        }
    }

    return 0;
}
